### PR TITLE
TEI to markdown improvements.

### PR DESCRIPTION
**Pull request by vvasuki**
tei-to-markdown-sarit.xsl can be omitted - it is specific to the sarit database.

### CODE REVIEW

**Comment by peterstadler** [comment]
@vvasuki  thank you for the PR! 
Since it's not up to date with our current dev branch and there's a merge conflict, may I ask you to rebase/update this PR so we can test it and eventually merge it?!

**Comment by vvasuki** [comment]
Fixed merge conflict. To aid comparison modulo formatting changes - https://www.diffchecker.com/lSqcB5r9 or https://i.imgur.com/5kXtV3b.png

File: markdown/tei-to-markdown.xsl [hunk 1]
```diff
@@ -84,173 +84,197 @@
...
+        <xsl:call-template name="newline"/>
+    </xsl:template>
+
+    <xsl:template match="p|trailer|lg">
+        <xsl:call-template name="newline"/>
+        <xsl:apply-templates/>
+        <xsl:call-template name="newline"/>
+    </xsl:template>
+
+    <xsl:template match="note/p|ref">
```
Comment (peterstadler) [inline markdown/tei-to-markdown.xsl]: This template rule `<xsl:template match="note/p|ref">` is not self explanatory to me. I would expect to wrap the whole `tei:note` into brackets, not every nested paragraph by itself? Second, a straight forward translation of `tei:ref[@target]` to would be `[Link text](URL)`.
So, while this code might fit your project encoding, I would hesitate to merge it into the TEIC's Stylesheets which aim to be as generic as possible.

**Review by peterstadler — CHANGES_REQUESTED** [review]
Thanks for updating this PR and all your work!
Yet, there seems to be an invisible(?) regression since your https://github.com/sanskrit-coders/Stylesheets/blob/6d2d6ed61dccc1a3b6791f20981df7d10aa4b89f/markdown/tei-to-markdown.xsl#L297 is missing a whitespace after the "*". 
You can see that regression in our GitHub action tests at e.g. https://github.com/TEIC/Stylesheets/runs/4131270793

File: markdown/tei-to-markdown.xsl (hunk 1 above)
Comment (vvasuki) [reply markdown/tei-to-markdown.xsl]: I have made it so. Please check. 

Attaching files which show the improvement.

[mdtest1.local.md](https://github.com/TEIC/Stylesheets/files/7494267/mdtest1.local.md)
[mdtest1.orig.local.md](https://github.com/TEIC/Stylesheets/files/7494268/mdtest1.orig.local.md)

**Comment by vvasuki** [comment]
(Fixed the regression in list items.)

**Comment by sydb** [comment]
@vvasuki —
Sorry this is taking so long. A few of us took a more careful look at this, and overall it looks great. But the change creates an error in the test procedure: the file expected-results/mdtest2.md is no longer what is expected. Can you fix that or would you like me to do that for you? (In which case you have to give me r/w access to your repo.)
Thank you!
P.S. Also probably a good idea to merge from TEIC/Stylesheets dev branch, again.

**Comment by vvasuki** [comment]
@sydb It would be great if you could make the necessary changes yourself.
reg. rw access to my repo - you should have received a github invitation.

**Comment by sydb** [comment]
Excellent. Update to Test/expected-results/ and Test2/expected-results/ made. This change now passes all tests in both Test/ and Test2/. Note that I did not merge in the current TEIC:dev into this repo (only because I have forgotten the syntax).

On quick look I think it safe to say that this PR is not perfect (is missing lots of stuff it _could_ do, and there might even be some problems), but it is **so** much better than what we have now, I think it should be accepted and merged as soon as can reasonably be done.

**Comment by HelenaSabel** [comment]
Tested locally. All tests passed thanks to change made per issue #560. 

However I cannot merge this PR (because the tests here have failed). Either @vvasuki pulls and merges the main Stylesheets repository to obtain the necessary file, or maybe @sydb can do it since he has push privileges in this fork. In any case, this is ready to merge

**Comment by sydb** [comment]
Per @HelenaSabel comment above, latest Makefile tweaks have been merged in. The failures in the tests are problems with Slack notification, so I agree with her, this can be merged at will.

**Pull request merged**